python/HeaderService/hutils.py

```diff
@@ -427,19 +322,20 @@ def write_fits(self, filename, dtype='random', naxis1=None, naxis2=None, btype='
                 fits.write(data, extname=extname, header=hdr)
         LOGGER.info("FITS write time:{}".format(elapsed_time(t0)))
 
-    def write_header(self, filename, delimiter='END', newline=False):
-
+    def write_header(self, filename):
         """
-        Writes the single header file using the strict FITS notation (newline=False)
-        or the more human readable (newline=True) with a delimiter for multiple HDU's
+        Writes single header file using the strict FITS format (i.e. empty HDUs,
+        write_mode='fits') or more human readable (write_mode='string') with a
+        delimiter for multiple HDU's
         """
-        if newline:
-            # Update header to a single string
-            self.string_header(delimiter=delimiter)
-            with open(filename, 'w') as fobj:
-                fobj.write(self.hstring)
+        t0 = time.time()
+        if self.write_mode == 'fits':
+            self.write_header_fits(filename)
+        elif self.write_mode == 'string':
+            self.write_header_string(filename)
         else:
-            t0 = time.time()
-            self.write_header_emptyHDU(filename)
-            LOGGER.info("Header write time:{}".format(elapsed_time(t0)))
+            LOGGER.info("ERROR: header write_mode:{} not recognized".format(self.write_mode))
```

Comment:
This seems like more of a `LOGGER.error` or `LOGGER.warning` rather than `LOGGER.info`.  Given this should never happen and given that you want a header written should it not raise?

Reply:
Yes, it should be `LOGGER.error(message)` followed by a `raise ValueError(msg)`